src/MongoDB/Manager.c

```diff
@@ -200,6 +200,15 @@ static void php_phongo_manager_prep_uri_options(zval *options TSRMLS_DC) /* {{{
 				continue;
 			}
 
+            if (!strcasecmp(ZSTR_VAL(string_key), MONGOC_URI_READPREFERENCE)) {
+				ZVAL_DEREF(option);
+				SEPARATE_ZVAL_NOREF(option);
+				if (Z_TYPE_P(option) != IS_STRING) {
+                    phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Unsupported readPreference type, a string must be given");
```

Comment:
Yea, I can see that [here](https://github.com/mongodb/mongo-php-driver/blob/master/php_phongo.c#L1116) , but I think that as mentioned at PHPC-887 that we need to throw some meaningful exception to the user to tell him about the invalid passed option, the issue with validating the option type -in case passing `MongoDB\Driver\ReadPreference` for example- with bson that bson will not check for zval values like objects.
it's a bit confusing, the example provided in that ticket was about unsupported types, also your reply [here](https://groups.google.com/forum/#!msg/mongodb-user/Cun9RklOsR0/DDDtGrh3DwAJ)